Sources/Commands/Utilities/PluginDelegate.swift

```diff
@@ -398,25 +398,34 @@ final class PluginDelegate: PluginInvocationDelegate {
             cacheBuildManifest: false
         )
 
-        // Find the target in the build operation's package graph; it's an error if we don't find it.
-        let packageGraph = try await buildSystem.getPackageGraph()
-        guard let target = packageGraph.module(for: targetName) else {
-            throw StringError("could not find a target named “\(targetName)”")
+        func lookupDescription(
+            for moduleName: String,
+            destination: BuildParameters.Destination
+        ) throws -> ModuleBuildDescription? {
+            try buildSystem.buildPlan.buildModules.first {
+                $0.module.name == moduleName && $0.buildParameters.destination == destination
+            }
         }
 
-        // FIXME: This is currently necessary because `target(for:destination:)` can
-        // produce a module that is targeting host when `targetName`` corresponds to
-        // a macro, plugin, or a test. Ideally we'd ask a build system for a`BuildSubset`
-        // and get the destination from there but there are other places that need
-        // refactoring in that way as well.
-        let buildParameters = if target.buildTriple == .tools {
-                try swiftCommandState.toolsBuildParameters
-            } else {
-                try swiftCommandState.productsBuildParameters
-            }
+        // FIXME: The name alone doesn't give us enough information to figure out what
+        // the destination is, this logic prefers "target" over "host" because that's
+        // historically how this was setup. Ideally we should be building for both "host"
+        // and "target" if module is configured for them but that would require changing
+        // `PluginInvocationSymbolGraphResult` to carry multiple directories.
+        let description: ModuleBuildDescription? = if let targetDescription = try lookupDescription(for: targetName, destination: .target) {
+            targetDescription
+        } else if let hostDescription = try lookupDescription(for: targetName, destination: .host) {
+            hostDescription
+        } else {
+            nil
```

Comment:
throw here?

Reply:
Cannot throw from if expression I think, I will check if that’s the case.